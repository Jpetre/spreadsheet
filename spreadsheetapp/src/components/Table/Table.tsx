import React from 'react';
import { env } from '../../config/config';
import Row from '../Row/Row';
import { CellDataType } from '../../../../types/types';

type TableProps = {
  x: number;
  y: number;
};

type TableState = {
  data: { [key: string]: CellDataType };
};

export const SpreadSheetContext = React.createContext({
  data: {},
  updateData: (data: { [key: string]: CellDataType }) => { }
});

export default class Table extends React.PureComponent<TableProps, TableState> {
  constructor(props: TableProps) {
    super(props);
    let data: { [key: string]: CellDataType } = {};
    for (let x = 1; x < this.props.x + 1; x++) {
      for (let y = 1; y < this.props.y + 1; y++) {
        data[x] = {
          ...data[x],
          [y]: {
            value: '',
            inputValue: '',
          }
        }
      }
    }

    this.state = {
      data,
    }
  };

  fetchComputations = (computations: string) =>
    fetch(`${env.API}${computations}`)
      .then(res => res.json());

  updateData = (data: { [key: string]: CellDataType }) => {
    console.log('data update', data);
    this.fetchComputations(JSON.stringify(data).replace(/\//g, '!'))
      .then((res: { result: { [key: string]: CellDataType } }) => {
        console.log(res.result);
        this.setState({
          data: res.result
        });
      })
      .catch(error => {
        console.log('error', error);
      });
  }

  render() {
    const rows = [];
    for (let y = 0; y < this.props.y + 1; y++) {
      rows.push(
        <Row
          key={y}
          y={y}
          x={this.props.x + 1}
        />,
      )
    };

    return (
      <div>
        <SpreadSheetContext.Provider
          value={{
            data: this.state.data,
            updateData: this.updateData
          }}
        >
          {rows}
        </SpreadSheetContext.Provider>
      </div>
    )
  }
};