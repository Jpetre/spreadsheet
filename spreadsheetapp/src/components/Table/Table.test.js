import React from 'react';
import Table from './Table';
import renderer from 'react-test-renderer';

test('Table render correctly', () => {
  const component = renderer.create(
    <Table x={6} y={6} id={'1'} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});