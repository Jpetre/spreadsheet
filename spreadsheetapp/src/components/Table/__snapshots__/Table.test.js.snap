// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render correctly 1`] = `
<div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      0
    </div>
    <div
      className="Cell Cell--info"
    >
      a
    </div>
    <div
      className="Cell Cell--info"
    >
      b
    </div>
    <div
      className="Cell Cell--info"
    >
      c
    </div>
    <div
      className="Cell Cell--info"
    >
      d
    </div>
    <div
      className="Cell Cell--info"
    >
      e
    </div>
    <div
      className="Cell Cell--info"
    >
      f
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      1
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      2
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      3
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      4
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      5
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="Row"
  >
    <div
      className="Cell Cell--info"
    >
      6
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="Cell"
      onClick={[Function]}
    >
      
    </div>
  </div>
</div>
`;
